source/structuring.rst

```diff
@@ -14,402 +14,545 @@ functions is better than copying-and-pasting duplicate bits of code
 
-For this example, let's say we want to define a customer record, where
-each customer may have both a shipping and a billing address.
-Addresses are always the same---they have a street address, city and
-state---so we don't want to duplicate that part of the schema
-everywhere we want to store an address.  Not only would that make the
-schema more verbose, but it makes updating it in the future more
-difficult.  If our imaginary company were to start doing international
-business in the future and we wanted to add a country field to all the
-addresses, it would be better to do this in a single place rather than
-everywhere that addresses are used.
+.. _schema-identification:
 
-So let's start with the schema that defines an address::
+Schema Identification
+---------------------
+
+Like any other code, schemas are easier to maintain if they can be
+broken down into logical units that reference each other as necessary.
+In order to reference a schema, we need a way to identify a schema.
+Schema documents are identified by non-relative URIs.
+
+.. note::
+   URI terminology can sometimes be unintuitive. In this document, the
+   following definitions are used.
+
+   - **URI** `[1]
+     <https://datatracker.ietf.org/doc/html/rfc3986#section-3>`__ or
+     **non-relative URI**: A
+     full URI containing a scheme (``https``) and a domain. It may
+     contain a URI fragment (``#foo``). Sometimes this document will
+     use "non-relative URI" to make it extra clear relative URIs are
+     not allowed.
+   - **relative reference** `[2]
+     <https://datatracker.ietf.org/doc/html/rfc3986#section-4.2>`__: A
+     partial URI that does not contain a scheme (``https``) and/or a
+     domain. It may contain a fragment (``#foo``).
+   - **URI-reference** `[3]
+     <https://datatracker.ietf.org/doc/html/rfc3986#section-4.1>`__: A
+     relative reference or non-relative URI. It may contain a URI
+     fragment (``#foo``).
+   - **absolute URI** `[4]
+     <https://datatracker.ietf.org/doc/html/rfc3986#section-4.3>`__ A
+     full URI containing a scheme (``https``) and a domain but not a
+     URI fragment (``#foo``)
+
+Schemas documents are not required to have an identifier, but
+obviously you will need one if you want to reference the schema from
+another. In this document, we will refer to schemas with no
+identifier as "anonymous schemas".
+
+In the following sections we will see how the "identifier" for a schema is
+determined.
+
+.. index::
+   single: retrieval URI
+   single: structuring; schema identification; retrieval URI
+
+.. _retrieval-uri:
+
+Retrieval URI
+~~~~~~~~~~~~~
+
+Even though schemas are identified by URIs, those identifiers are not
+necessarily network accessible. They are just identifiers.
+Implementations generally provide a way for you to load schemas into
+an internal schema database and they retrieve schemas from there when
+they are referenced.
+
+How ever the schema is retrieved, the non-relative URI used to fetch
```

Comment:
I thought this might be confusing. I wasn't intending to use the conjunctive adverb "however". I meant it like "no matter how", which I believe is not spelled the same as the conjunctive adverb. I think this wording would be fine if it was spoken, but when reading it, it's distracting because it looks like it's misspelled. Maybe this will be better,

```suggestion
No matter how the schema is retrieved, the non-relative URI used to fetch
```